### PR TITLE
feat: Add AddSubscription to Subscriptions.razor

### LINKED ISSUE #28: Add a + button for creating a new subscription
- Use the /docs/MVP mockups.png as reference
- A **"+"** button should be visible on the bottom right of the start and monthly page

### Suggested solution
- https://icons.getbootstrap.com/icons/plus-circle/

### Acceptance criteria
- Looks good on a ~1080x2400 (google pixel 5) resolution phone
- A png of the page is attached to the Pull request (not the repository code!) so that others don't need to run the code locally

### PR DESCRIPTION
**Pull request by bigggmoustache**
Added Subscriptions.razor to hold AddSubscriptionsButton component.
Added Components folder to Shared folder since component is to be shared.

image

### CODE REVIEW

**Comment by bigggmoustache** [comment]
I will add sample content to scroll over and make the button float over it today.

**Comment by chrisK00** [comment]
Make the button bigger and just a tiny bit more to the left.

File: subtrack.MAUI/Pages/Subscriptions.razor [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿@using Components
+@page "/subscriptions"
+
+<h3>Subscriptions</h3>
+
+<div style="
```
Comment (chrisK00) [inline subtrack.MAUI/Pages/Subscriptions.razor]: place this inside the button component instead. also add it to the startpage

File: subtrack.MAUI/Shared/Components/AddSubscriptionButton.razor [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿@namespace Components
+<div style="
+    ">
+    <NavLink class="nav-link d-flex align-items-center" data-bs-theme="dark" href="/">
```
Comment (chrisK00) [inline subtrack.MAUI/Shared/Components/AddSubscriptionButton.razor]: href should be /create or /new

**Review by chrisK00 — CHANGES_REQUESTED** [review]

**Comment by chrisK00** [comment]
> 

Im gonna merge #27 once the build succeeds so you can just use that instead

File: subtrack.MAUI/Pages/Subscriptions.razor (hunk 1 above)
Comment (bigggmoustache) [reply subtrack.MAUI/Pages/Subscriptions.razor]: Moved the CSS as requested and added a z-index.

**Comment by bigggmoustache** [comment]
would it make sense to have this button send a returnurl to "CreateSubscription" page?

File: subtrack.MAUI/Shared/Components/AddSubscriptionButton.razor (hunk 2 above)
Comment (bigggmoustache) [reply subtrack.MAUI/Shared/Components/AddSubscriptionButton.razor]: href updated to have / at start.

**Comment by chrisK00** [comment]
> would it make sense to have this button send a returnurl to "CreateSubscription" page?

Yes thats what we are going to have to do. Because this button is shared  Add a parameter named return url. You can look at SubscriptionMonthIyem.razor to see how a param is added and then Monthly.razor utilized it

**Comment by bigggmoustache** [comment]
> > would it make sense to have this button send a returnurl to "CreateSubscription" page?
> 
> Yes thats what we are going to have to do. Because this button is shared Add a parameter named return url. You can look at SubscriptionMonthIyem.razor to see how a param is added and then Monthly.razor utilized it

I was thinking a base class for functional components holding frequently used properties like returnurl would be nice? Components that navigate etc

Edit: I didn't mean to close sorry!

**Comment by chrisK00** [comment]
> > > would it make sense to have this button send a returnurl to "CreateSubscription" page?
> > 
> > Yes thats what we are going to have to do. Because this button is shared Add a parameter named return url. You can look at SubscriptionMonthIyem.razor to see how a param is added and then Monthly.razor utilized it
> 
> I was thinking a base class for functional components holding frequently used properties like returnurl would be nice? Components that navigate etc
> 
> Edit: I didn't mean to close sorry!

Just a prop is preferred over inheritance, its just a few loc anyway

**Comment by chrisK00** [comment]
How does it look in the UI now with the larger btn?

**Comment by bigggmoustache** [comment]
> 

image

Also I don't understand what's applicable about the monthlysubscription example you gave. I feel like I'm just having some brainfog here. My understanding is I need to pass the returnUrl from one component page to the next component page, where your example involves embedded components? Sorry for any misunderstanding.

**Comment by chrisK00** [comment]
> > 
> 
> image
> 
> Also I don't understand what's applicable about the monthlysubscription example you gave. I feel like I'm just having some brainfog here. My understanding is I need to pass the returnUrl from one component page to the next component page, where your example involves embedded components? Sorry for any misunderstanding.

Yeah you are correct, this is what we want  https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-7.0#query-strings

**Comment by bigggmoustache** [comment]
> > > 
> > 
> > 
> > image
> > Also I don't understand what's applicable about the monthlysubscription example you gave. I feel like I'm just having some brainfog here. My understanding is I need to pass the returnUrl from one component page to the next component page, where your example involves embedded components? Sorry for any misunderstanding.
> 
> Yeah you are correct, this is what we want https://learn.microsoft.com/en-us/aspnet/core/blazor/fundamentals/routing?view=aspnetcore-7.0#query-strings

Okie doke. Just making sure I didn't misunderstand before committing again! Will submit in a bit.

**Review by chrisK00 — APPROVED** [review]
Merge ready ✅

**Pull request merged**